src/Simplex/Messaging/Agent.hs

```diff
@@ -1053,25 +1059,25 @@ runCommandProcessing c@AgentClient {subQ} server_ = do
 enqueueMessages :: AgentMonad m => AgentClient -> ConnData -> NonEmpty SndQueue -> MsgFlags -> AMessage -> m AgentMsgId
 enqueueMessages c cData sqs msgFlags aMessage = do
   when (ratchetSyncSendProhibited cData) $ throwError $ INTERNAL "enqueueMessages: ratchet is not synchronized"
-  enqueueMessages' c cData sqs msgFlags aMessage
+  execAgentBatch c $ \b -> enqueueMessages' c b cData sqs msgFlags aMessage
 
-enqueueMessages' :: AgentMonad m => AgentClient -> ConnData -> NonEmpty SndQueue -> MsgFlags -> AMessage -> m AgentMsgId
-enqueueMessages' c cData (sq :| sqs) msgFlags aMessage = do
-  msgId <- enqueueMessage c cData sq msgFlags aMessage
-  mapM_ (enqueueSavedMessage c cData msgId) $
-    filter (\SndQueue {status} -> status == Secured || status == Active) sqs
+enqueueMessages' :: AgentMonad' m => AgentClient -> AgentBatch m -> ConnData -> NonEmpty SndQueue -> MsgFlags -> AMessage -> BatchT AgentErrorType m AgentMsgId
+enqueueMessages' c b cData (sq :| sqs) msgFlags aMessage = do
+  msgId <- enqueueMessage c b cData sq msgFlags aMessage
+  mapM_ (enqueueSavedMessage c b cData msgId) $ -- TODO: use batching
+    filter isActiveSndQ sqs
   pure msgId
 
-enqueueMessage :: forall m. AgentMonad m => AgentClient -> ConnData -> SndQueue -> MsgFlags -> AMessage -> m AgentMsgId
-enqueueMessage c cData@ConnData {connId} sq msgFlags aMessage = do
-  resumeMsgDelivery c cData sq
-  aVRange <- asks $ smpAgentVRange . config
-  msgId <- storeSentMsg $ maxVersion aVRange
-  queuePendingMsgs c sq [msgId]
+enqueueMessage :: AgentMonad' m => AgentClient -> AgentBatch m -> ConnData -> SndQueue -> MsgFlags -> AMessage -> BatchT AgentErrorType m AgentMsgId
```

Comment:
same for this, if this new one is called enqueueMessageB, the old one would stay, and some calls won't change